### ISSUE TITLE
regression_RobustAutoencoder not implemented

### ISSUE THREAD

**Issue by mlix11** [raghavchalapathy/rcae#1]
I read your paper and I liked it a lot. So I want to try the things out on my own. 

But I have one Issue, where do you get the **tflearn.regression_RobustAutoencoder** function,  because it seams it is not implemented in tflearn.  Is it self-implemented or on a branch/fork of tflearn?

For example in _section_5.1_anomaly_detection_CIFAR_10_AE.py_:
```python
mue = 0.1
net = tflearn.regression_RobustAutoencoder(net,mue,hidden_layer,decoder, optimizer='adam', learning_rate=0.001,
                        loss='rPCA_autoencoderLoss_FobsquareLoss', metric=None,name="vanilla_autoencoder")

model = tflearn.DNN(net, tensorboard_verbose=0)
```

Thank you!

**Comment by maryam089** [comment]
Well I have also read this paper and liked the idea so when i tried to implement the codes in python, I found the similar issue that tflearn.regression_RobustAutoencoder function,  is not implemented in tflearn. 
How to deal with this issue ? @raghavchalapathy 
Are you able to resolve that issue you mentioned above ? @loose11

**Comment by raghavchalapathy** [comment]
Its is self implemented essentially only the parameters essential for defining loss function is added


def rPCA_autoencoderLoss_FobsquareLoss(y_pred, y_true,mue,hidden_layer,decode_layer):
    """ Mean Square Loss.

    Arguments:
        y_pred: `Tensor` of `float` type. Predicted values.
        y_true: `Tensor` of `float` type. Targets (labels).

    """
    with tf.name_scope("MeanSquare"):
        return tf.reduce_mean(tf.square(y_pred - y_true))

**Comment by maryam089** [comment]
Kindly update it in your code to remove that bug.
Cheers thanks :) @raghavchalapathy

**Comment by tayden** [comment]
@raghavchalapathy Can you please give more code for the loss function used?

**Comment by AdamRain** [comment]
Same question...... Can you please kindly upload the **regression_RobustAutoencoder** function?

**Comment by raghavchalapathy** [comment]
Hello All

 I have updated the source code repository and used Keras/ tensorflow
You may see the latest repository here ( This includes RCAE: Robust convolution autoencoder)
https://github.com/raghavchalapathy/oc-nn/blob/master/README.md
from src.models.RCAE import RCAE_AD

with regards
Raghav

**Comment by raghavchalapathy** [comment]
from src.models.RCAE import RCAE_AD

**Comment by jianzhuwang** [comment]
Is there a complete code for restaurant dataset? I did not see the detailed code for that dataset, and I failed to know the network structure and other information. How to obtain the foreground? Also, I have the same question as posted above. 

Many thanks for your help.
